src/lib/tflm/tflm_backend.cc

```diff
@@ -192,38 +193,58 @@ static void ml_backend_init_ops_resolver(MicroOpsResolver *resolver) {
 int ml_backend_init_model(py_ml_model_obj_t *model) {
     RegisterDebugLogCallback(ml_backend_log_handler);
 
-    // Parse model's data.
+    // Parse the model's data.
     const Model *tflite_model = GetModel(model->data);
     if (tflite_model->version() != TFLITE_SCHEMA_VERSION) {
         mp_raise_msg(&mp_type_ValueError, MP_ERROR_TEXT("Unsupported model schema"));
     }
 
-    // Initialize the op resolver.
+    // Initialize a temporary op resolver.
     MicroOpsResolver resolver;
     ml_backend_init_ops_resolver(&resolver);
 
-    // Allocate the interpreter and tensors once to initialize the model, check input
-    // and output data types and to get the optimal tensor arena size.
-    fb_alloc_mark();
-    uint32_t tensor_arena_size;
-    uint8_t *tensor_arena = (uint8_t *) fb_alloc_all(&tensor_arena_size, FB_ALLOC_PREFER_SIZE | FB_ALLOC_CACHE_ALIGN);
-
-    MicroInterpreter interpreter(tflite_model,
-                                 resolver,
-                                 tensor_arena,
-                                 tensor_arena_size);
+    gc_info_t info;
+    gc_info(&info);
+    // Allocate a temporary interpreter to get the optimal arena size.
+    size_t arena_size = info.max_free - TF_ARENA_ALIGN;
+    uint8_t *arena_memory = m_new(uint8_t, arena_size);
+    MicroInterpreter interpreter(tflite_model, resolver, arena_memory, arena_size);
```

Comment:
Don't you need to align the initial tensor arena here before ou pass it?

Reply:
Either way works, you can align the pointer or round up the returned size.

Reply:
I've updated the code to simplify this process. The library seems to align the pointer internally, if needed, so the arena size must be big enough for `size + alignment`. If not, the first pass will fail, as there's not enough GC memory for size + alignment. If it succeeds then the rounded up `used_bytes()` can be used for the second pass with or without an aligned pointer.

Reply:
Works great.